### PR TITLE
Document edgedb.toml

### PR DESCRIPTION
**Pull request by raddevon**

### CODE REVIEW

File: docs/cli/edgedb_toml.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- ``server-version``- The server version of the EdgeDB project.
+
+  .. note::
+
+      The version specification is assumed to be **a minimum version**, but the
+      CLI will *not* upgrade to subsequent major versions. This means if the
+      version specified is ``3.1`` and versions 3.2 and 3.3 are available, 3.3
+      will be installed, even if version 4.0 is also available.
+
+      To specify an exact version, prepend with ``=`` like this: ``=3.1``
```
Comment (scotttrinh) [inline docs/cli/edgedb_toml.rst:25]: I think we support the full range of Rust cargo-flavored semver specifiers? Do we want to point that out and link to some reference material like https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html?

File: docs/cli/edgedb_toml.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
+.. _ref_cli_edgedb_toml:
+
+===========
+edgedb.toml
+===========
+
+The ``edgedb.toml`` file is a key component of EdgeDB's project system. If this
```
Comment (scotttrinh) [inline docs/cli/edgedb_toml.rst:7]: Do you think it's helpful to link out to any TOML format documentation like https://toml.io/en/v1.0.0 ?

File: docs/cli/edgedb_toml.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+file is present in a directory, it signals to the CLI that the directory is an
```
Comment (fantix) [inline docs/cli/edgedb_toml.rst:8]: > it signals to the CLI

also to any client bindings ;)

Comment (scotttrinh) [reply docs/cli/edgedb_toml.rst:8]: Oh, good point: I wonder if this should live outside of the CLI docs? Or at least also live somewhere about the general project structure of an EdgeDB project?

Comment (raddevon) [reply docs/cli/edgedb_toml.rst:8]: I almost put it in the reference section instead, but I thought only the CLI cared about it. If client bindings also use it, I think it makes sense to put it there instead. Thanks, @fantix!

**Review by elprans — APPROVED** [review]

File: docs/cli/edgedb_toml.rst (hunk 2 above)
Comment (raddevon) [reply docs/cli/edgedb_toml.rst]: I found a note with a link to TOML docs in the projects reference, so I just copied that one over. I did like your target better, so I changed it from targeting the TOML quickstart to what you suggested.

**Pull request merged**